pkg/clusterapi/upgrader.go

```diff
@@ -63,7 +63,7 @@ func (c *CAPIChangeDiff) toChangeDiff() *types.ChangeDiff {
 	return types.NewChangeDiff(r...)
 }
 
-func (u *Upgrader) capiChangeDiff(currentSpec, newSpec *cluster.Spec, provider providers.Provider) *CAPIChangeDiff {
+func CapiChangeDiff(currentSpec, newSpec *cluster.Spec, provider providers.Provider) *CAPIChangeDiff {
```

Comment:
Same here, I would vote to return `types.ChangeDiff`

Reply:
Will update this in the cr where I display the CAPI components too.